external/vaults/models.go

```diff
@@ -359,26 +359,76 @@ func ApplyKongData(externalVault *TExternalVault, vault models.TVault) {
 		return
 	}
 
-	// Apply kong fees with priority: hook > snapshot > vault model
-	managementFee := kongData.Snapshot.ManagementFee.Value
-	performanceFee := kongData.Snapshot.PerformanceFee.Value
-
-	// Override with hook fees if they exist (hook takes precedence)
-	if kongData.Hook.Fees.ManagementFee != 0 {
+	managementFee := kongData.Fees.ManagementFee
+	performanceFee := kongData.Fees.PerformanceFee
+	if managementFee == 0 {
 		managementFee = kongData.Hook.Fees.ManagementFee
 	}
-	if kongData.Hook.Fees.PerformanceFee != 0 {
+	if performanceFee == 0 {
 		performanceFee = kongData.Hook.Fees.PerformanceFee
 	}
+	if managementFee == 0 {
+		managementFee = kongData.Snapshot.ManagementFee.Value
+	}
+	if performanceFee == 0 {
+		performanceFee = kongData.Snapshot.PerformanceFee.Value
+	}
 
-	// Convert from basis points to percentage and apply to APR fees
 	externalVault.APR.Fees.Management = bigNumber.NewFloat(float64(managementFee) / 10000.0)
 	externalVault.APR.Fees.Performance = bigNumber.NewFloat(float64(performanceFee) / 10000.0)
 
-	// Future: Apply other kong data
-	// applyKongPrices(externalVault, kongData)
-	// applyKongAPY(externalVault, kongData)
-	// applyKongMetadata(externalVault, kongData)
+	if kongData.TotalAssets != nil {
+		externalVault.TVL.TotalAssets = kongData.TotalAssets
+	}
+	if kongData.TVL.Close > 0 {
```

Comment:
-- don't use multicalls as fallbacks if kong is missing data
if kong is missing data it's an error. flag it and invetigate. this applies in particular to tvl. if kong doesn't have a tvl, it's an error and needs investigating. note, this was specified wrong in the original issue that mentioned using a fallback for tvl. original issue has been updated.